libs/agno/agno/run/workflow.py

```diff
@@ -0,0 +1,157 @@
+from dataclasses import asdict, dataclass, field
+from enum import Enum
+from time import time
+from typing import Any, Dict, List, Optional, Union
+
+from pydantic import BaseModel
+
+from agno.media import AudioArtifact, AudioResponse, ImageArtifact, VideoArtifact
+from agno.models.message import Message
+from agno.run.response import RunResponse
+from agno.run.team import TeamRunResponse
+
+
+class WorkflowRunEvent(str, Enum):
+    """Events that can be sent by workflow execution"""
+
+    workflow_started = "WorkflowStarted"
+    workflow_completed = "WorkflowCompleted"
+    workflow_error = "WorkflowError"
+
+    task_started = "TaskStarted"
+    task_completed = "TaskCompleted"
+    task_error = "TaskError"
+
+
+@dataclass
+class WorkflowRunResponse:
+    """Response returned by Workflow.run() functions"""
+
+    event: str = WorkflowRunEvent.task_completed.value
+
+    content: Optional[Any] = None
+    content_type: str = "str"
+    messages: Optional[List[Message]] = None
+    metrics: Optional[Dict[str, Any]] = None
+
+    # Workflow-specific fields
+    workflow_id: Optional[str] = None
+    workflow_name: Optional[str] = None
+    sequence_name: Optional[str] = None
+    task_name: Optional[str] = None
+    task_index: Optional[int] = None
```

Comment:
single task?

Reply:
for events, like when we get `task_started`, the workflow run response stores the name of and index of that task. I use it in the cookbook- `cookbook/workflows/capture_workflow_events.py`

can remove if not needed.

```py
 elif response.event == WorkflowRunEvent.task_started:
                task_name = response.task_name or "Unknown"
```